pkg/capr/planner/certificaterotation.go

```diff
@@ -139,20 +142,81 @@ func (p *Planner) rotateCertificatesPlan(controlPlane *rkev1.RKEControlPlane, to
 		Content: base64.StdEncoding.EncodeToString([]byte(idempotentRotateScript)),
 		Path:    rotateScriptPath,
 	})
-	rotatePlan.Instructions = append(rotatePlan.Instructions, []plan.OneTimeInstruction{
-		{
-			Name:    "rotate certificates",
-			Command: "sh",
-			Args:    args,
+	rotatePlan.Instructions = append(rotatePlan.Instructions, plan.OneTimeInstruction{
+		Name:    "rotate certificates",
+		Command: "sh",
+		Args:    args,
+	})
+	if isControlPlane(entry) {
+		if kcmCertDir := getArgValue(config[KubeControllerManagerArg], CertDirArgument, "="); kcmCertDir != "" && getArgValue(config[KubeControllerManagerArg], TLSCertFileArgument, "=") == "" {
```

Comment:
it looks like we will always delete the kube-scheduler's cert and key.  Should we check if the kube-controller-manager is in the list of `rotation.Services`?  Same question to Kube-scheduler.

Reply:
Given the fact that RKE2 doesn't manage these certificates at all, I don't think we should limit certificate rotation to the kube-scheduler and/or kube-controller-manager being specified. These certificates are only used by the CAPR Planner and should always be rotated when possible to ensure maximum length of validity.

The `service` arguments are passed to `rke2` and the engine service couldn't care less whether we are manipulating the service certificates for the healthz endpoint.